src/pushjack/apns.py

```diff
@@ -764,6 +776,346 @@ class APNSExpiredToken(namedtuple('APNSExpiredToken', ['token', 'timestamp'])):
+                together in Notification Center. Defaults to ``None``.
+            extra (dict, optional): Extra data to include with the alert.
+        """
+        assert device_id, 'Must provide device IDs.'
+
+        if max_payload_length is None:
+            max_payload_length = self.default_max_payload_length
+
+        message = APNSMessage(message,
+                              max_payload_length=max_payload_length,
+                              **options)
+
+        validate_tokens([device_id])
+        validate_message(message)
+
+        return self._send_message(
+            device_id=device_id,
+            message=message,
+            low_priority=low_priority,
+            expiration=expiration)
+
+    def send_messages(self,
+                      device_ids,
+                      message=None,
+                      expiration=None,
+                      low_priority=None,
+                      max_payload_length=None,
+                      **options):
+        unsendable_devices = []
+        if max_payload_length is None:
+            max_payload_length = self.default_max_payload_length
+
+        message = APNSMessage(message,
+                              max_payload_length=max_payload_length,
+                              **options)
+
+        validate_tokens(device_ids)
+        validate_message(message)
+
+        for device_id in device_ids:
+
+            try:
+                self._send_message(
+                    device_id=device_id, message=message,
+                    low_priority=low_priority, expiration=expiration)
+            except Exception:
+                unsendable_devices.append(device_id)
+
+        if not unsendable_devices:
+            return
+
+        if len(unsendable_devices) != len(device_ids):
+            raise APNSMessagePartiallySentError
+
+        raise APNSUnsendableError
+
+    def _send_message(
+            self, device_id, message, low_priority,
+            identifier=None, expiration=None):
+        json_data = message.to_json()
+
+        if low_priority:
+            priority = str(APNS_LOW_PRIORITY)
+        else:
+            priority = str(APNS_HIGH_PRIORITY)
+
+        if expiration is None:
+            expiration = str(int(time.time() + self.default_expiration_offset))
+
+        if not identifier:
+            identifier = str(uuid.uuid4())
+
+        return self._post(device_id, json_data,
+                          priority, expiration, identifier)
+
+    def _post(self, device_id, data, priority, expiration, identifier):
+        request_headers = {
+            'apns-expiration': expiration,
+            'apns-priority': priority,
+            'apns-id': identifier,
+            'apns-topic': self.bundle_id,
+            'authorization': 'bearer {0}'.format(self.token.encrypted_token)
+        }
+
+        try:
+            steam_id = self.conn.request(
+                method='POST',
+                url='/3/device/{0}'.format(device_id),
```

Comment:
Should probably make the URL a constant.

Reply:
Yes, it would be better if the URL is a constant.